### PR TITLE
Add Shape map to BCO domains

### LINKED ISSUE #4764: Add shape map to BBH domains
Task list:

- [x] Add distorted frames to BBH domains, not changing any maps
- [ ] Add shape map -- PRs for single and binary domains

Both the cubical and cylindrical BBH domains should be updated in parallel.

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

Fixes #4764

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

~Depends on and includes #4824, #4844, #4846, #4847.~

### CODE REVIEW

**Comment by knelli2** [comment]
Ok this PR is finally ready now.

File: src/Domain/Creators/BinaryCompactObjectHelpers.cpp [hunk 1]
```diff
@@ -129,76 +129,71 @@ void TimeDependentMapOptions::build_maps(
...
+             Object == domain::ObjectLabel::None,
+         "Object label for TimeDependentMapOptions must be either A, B or "
+         "None, not"
              << Object);
-  return Object == domain::ObjectLabel::A ? 0_st : 1_st;
+  return Object == domain::ObjectLabel::A
+             ? 0_st
+             : (Object == domain::ObjectLabel::B
+                    ? 1_st
+                    : std::numeric_limits<size_t>::max());
```
Comment (nilsvu) [inline src/Domain/Creators/BinaryCompactObjectHelpers.cpp]: You don't/shouldn't call this function with `ObjectLabel::None`, right? Then keep the assert and the condition here unchanged.

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/Domain/Creators/BinaryCompactObjectHelpers.cpp (hunk 1 above)
Comment (knelli2) [reply src/Domain/Creators/BinaryCompactObjectHelpers.cpp]: Yeah I guess because I just `ASSERT` and don't `static_assert` that would work.

**Comment by knelli2** [comment]
Ok rebased and squashed.

**Review by nilsvu — DISMISSED** [review]

**Comment by nilsvu** [comment]
gcc want's you to do `noreturn` things

**Comment by knelli2** [comment]
I didn't feel like trying to figure out how to do `[[noreturn]]` for only one template so I just ignored the suggestion for gcc

**Review by nilsvu — DISMISSED** [review]

**Comment by knelli2** [comment]
Ok screw templates. I just moved it to runtime because clang-tidy was unhappy now. This doesn't run that often either so it doesn't matter.

**Review by nilsvu — DISMISSED** [review]

**Review by nilsvu — APPROVED** [review]

**Pull request merged**